File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/changes/HowToSubscribeToReplicationConflictsChanges.java
package net.ravendb.clientapi.changes;

import java.io.Closeable;

import net.ravendb.abstractions.closure.Action1;
import net.ravendb.abstractions.data.ReplicationConflictNotification;
import net.ravendb.abstractions.data.ReplicationConflictTypes;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.changes.IObservable;
import net.ravendb.client.document.DocumentStore;
import net.ravendb.client.utils.Observers;


public class HowToSubscribeToReplicationConflictsChanges {

  @SuppressWarnings("unused")
  private interface IFoo {
    //region replication_conflicts_changes_1
    public IObservable<ReplicationConflictNotification> forAllReplicationConflicts();
    //endregion
  }

  @SuppressWarnings("unused")
  public HowToSubscribeToReplicationConflictsChanges() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      //region replication_conflicts_changes_2
      Closeable subscription = store
        .changes()
        .forAllReplicationConflicts()
        .subscribe(Observers.create(new Action1<ReplicationConflictNotification>() {
          @Override
          public void apply(ReplicationConflictNotification conflict) {
            if (ReplicationConflictTypes.DOCUMENT_REPLICATION_CONFLICT.equals(conflict.getItemType())) {
              System.out.println("Conflict detected for " + conflict.getId());
            }
          }
        }));
      //endregion
    }
  }
}

File: Documentation/3.5/Samples/csharp/Raven.Documentation.Samples/ClientApi/Changes/HowToSubscribeToReplicationConflictsChanges.cs
using System;

using Raven.Abstractions.Data;
using Raven.Client.Changes;
using Raven.Client.Document;

namespace Raven.Documentation.Samples.ClientApi.Changes
{
	public class HowToSubscribeToReplicationConflictsChanges
	{
		private interface IFoo
		{
			#region replication_conflicts_changes_1
			IObservableWithTask<ReplicationConflictNotification>
				ForAllReplicationConflicts();
			#endregion
		}

		public HowToSubscribeToReplicationConflictsChanges()
		{
			using (var store = new DocumentStore())
			{
				#region replication_conflicts_changes_2
				IDisposable subscription = store.Changes()
					.ForAllReplicationConflicts()
					.Subscribe(conflict =>
					{
						if (conflict.ItemType == ReplicationConflictTypes.DocumentReplicationConflict)
						{
							Console.WriteLine("Conflict detected for {0}. Ids of conflicted docs: {1}. " +
											  "Type of replication operation: {2}",
											  conflict.Id,
											  string.Join(", ", conflict.Conflicts),
											  conflict.OperationType);
						}
					});
				#endregion
			}
		}
	}
}
